### PR TITLE
feat: finer-wallet support

### PR DESCRIPTION
**Pull request by prih**
# Description

Add FiNER Wallet App support.

FiNER Wallet implements an API almost identical to near-wallet over Deep linking on IOS and Android.

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

# Type of change. This type of change is the main reason for the PR.

- [x] FEATURE - a PR of this type introduces a new feature.

# Breaking changes

- [x] NO BREAKING CHANGE - this PR doesn't contain any breaking changes and it's backwards compatible

### CODE REVIEW

**Comment by amirsaran3** [comment]
Hey @prih.

I cloned your repo and checked out the `fintoda:dev` branch. I wanted to test Finer wallet before I can approve this PR. When I tried to connect with Finer Wallet it was not working. I got this error:
```
Failed to launch 'finer://wallet.testnet.near.org/login/?success_url=http%3A%2F%2Flocalhost%3A4200%2Fwallet-selector&failure_url=http%3A%2F%2Flocalhost%3A4200%2Fwallet-selector&contract_id=guest-book.testnet&public_key=ed25519%3A4Tt9pYDsFpnhTeDEs8Ch4PioWFfLc5he4qKZ7n2EJtsP' because the scheme does not have a registered handler.
```

My questions:
1. Is there a link to more information about Finer Wallet? I only found this link: https://finerwallet.io/.
2. Can you help me understand how I can connect to the Finer Wallet in your branch so I can test?

**Comment by amirsaran3** [comment]
> > 
> 
> @amirsaran3 hi,
> 
> finer:// scheme is available on the mobile platform, Finer is a mobile application. On a mobile operating system, an application can assign a schema to itself.

Thanks for the reply @prih. I can't find the finer wallet in the Play Store. 

How can I test the implementation of the finer wallet while developing?

File: packages/finer-wallet/src/lib/finer-wallet.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  return async (options) => {
+    const wallet = await setupMyNearWallet({
+      walletUrl: resolveWalletUrl(options.options.network, walletUrl),
+      iconUrl,
+    })(options);
+
+    if (!wallet) {
+      return null;
+    }
+
```
Comment (amirsaran3) [inline packages/finer-wallet/src/lib/finer-wallet.ts:42]: There should be a check with `is-mobile` to only show Finer Wallet on mobile devices.

File: packages/finer-wallet/src/lib/finer-wallet.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    return {
+      ...wallet,
+      id: "finer-wallet",
+      metadata: {
+        ...wallet.metadata,
+        name: "FiNER Wallet",
+        description: null,
+        iconUrl,
+        deprecated,
+        available: true,
```
Comment (amirsaran3) [inline packages/finer-wallet/src/lib/finer-wallet.ts:52]: `available` should be false if Finer Wallet is not installed.

**Comment by prih** [comment]
> > > 
> > 
> > 
> > @amirsaran3 hi,
> > finer:// scheme is available on the mobile platform, Finer is a mobile application. On a mobile operating system, an application can assign a schema to itself.
> 
> Thanks for the reply @prih. I can't find the finer wallet in the Play Store.
> 
> How can I test the implementation of the finer wallet while developing?

Finer is only available on iOS App Store (we are currently working on delivery to the Google Play Market)

File: packages/finer-wallet/src/lib/finer-wallet.ts [hunk 3]
```diff
@@ -49,7 +50,7 @@ export function setupFinerWallet({
         description: null,
         iconUrl,
         deprecated,
-        available: true,
+        available: isMobile(),
```
Comment (amirsaran3) [inline packages/finer-wallet/src/lib/finer-wallet.ts:53]: This is not quite right. `available` means if a wallet is installed and ready to be used with. For example injected wallets will have `available` false if the extension is not installed for that particular wallet.

`isMobile()` should be checked above and will `return null` if is not mobile.

File: packages/finer-wallet/src/lib/finer-wallet.ts [hunk 4]
```diff
@@ -50,7 +52,7 @@ export function setupFinerWallet({
         description: null,
         iconUrl,
         deprecated,
-        available: isMobile(),
+        available: true,
```
Comment (amirsaran3) [inline packages/finer-wallet/src/lib/finer-wallet.ts:55]: What will happen if Finer Wallet application is not installed and you try to connect?

Comment (prih) [reply packages/finer-wallet/src/lib/finer-wallet.ts:55]: in mobile browser - nothing will happen

Comment (prih) [reply packages/finer-wallet/src/lib/finer-wallet.ts:55]: I was thinking, we can make communication through the https:// scheme, and if the application is not installed, then the user will be taken to a website where he can proceed to install the application on the App Store

Comment (amirsaran3) [reply packages/finer-wallet/src/lib/finer-wallet.ts:55]: Currently this is how it looks when an injected wallet is not installed.

image

This screen will show if `available` is false. The Open button will redirect you to `downloadUrl` (your app download page)

**Comment by prih** [comment]
> Hey @prih. I checked out your latest changes and it is not quite the intended behavior. There is no "Not Installed" flow, it immediately redirects to your website.
> 
> Have you tried using deferred deep linking to redirect to app store if app is not installed?

We have no way to track if the app is installed or not, it's not a desktop web and it's not a browser extension. We have implemented support for communication through Universal Links (https://developer.apple.com/ios/universal-links/), the user gets to an intermediate page on the site, if he has the application installed, then the browser will offer to open the current url in the Finer application, also, if the browser is different from Safari, the page has a code that tries to follow the current link by changing the current scheme from https:// to finer://, and if the user has the application installed, then a transition will occur. In other cases, the user remains on the page where the link to go to the App Store is located. If you have a better idea, please suggest)

**Comment by AmmarHumackicSQA** [comment]
Hey @prih as a reminder, the security statement needs to be shown either on your GitHub repo or the website. You can check it [here](https://github.com/near/wallet-selector/blob/main/CONTRIBUTING.md#:~:text=GitHub%20account%20or%20Website) again. Thanks!

**Comment by prih** [comment]
> 

yes, confirm

**Comment by AmmarHumackicSQA** [comment]
> > 
> 
> yes, confirm

Can we get a link to it? We couldn't find it on your website. The security statement needs to be shown either on your GitHub repo or the website ([explained here](https://github.com/near/wallet-selector/blob/main/CONTRIBUTING.md))

**Comment by prih** [comment]
> > > 
> > 
> > 
> > yes, confirm
> 
> Can we get a link to it? We couldn't find it on your website. The security statement needs to be shown either on your GitHub repo or the website ([explained here](https://github.com/near/wallet-selector/blob/main/CONTRIBUTING.md))

we are actively developing the site and the support system, I will soon be able to provide. Main domain is https://finerwallet.io

**Comment by amirsaran3** [comment]
@prih There are conflicts right now. Can you resolve them when you have time and we can merge this PR.

**Comment by AmmarHumackicSQA** [comment]
> > > > 
> > > 
> > > 
> > > yes, confirm
> > 
> > 
> > Can we get a link to it? We couldn't find it on your website. The security statement needs to be shown either on your GitHub repo or the website ([explained here](https://github.com/near/wallet-selector/blob/main/CONTRIBUTING.md))
> 
> we are actively developing the site and the support system, I will soon be able to provide. Main domain is https://finerwallet.io

Hi @prih any news on this? Thanks.

**Comment by AmmarHumackicSQA** [comment]
> > > > 
> > > 
> > > 
> > > yes, confirm
> > 
> > 
> > Can we get a link to it? We couldn't find it on your website. The security statement needs to be shown either on your GitHub repo or the website ([explained here](https://github.com/near/wallet-selector/blob/main/CONTRIBUTING.md))
> 
> we are actively developing the site and the support system, I will soon be able to provide. Main domain is https://finerwallet.io

@prih pinging to check the status of this, and to see if this PR could be potentially included in our next release of Wallet Selector. Thanks!

**Comment by prih** [comment]
> 

we are working on resolving this issue

**Comment by AmmarHumackicSQA** [comment]
> > 
> 
> we are working on resolving this issue

Hey @prih any news on the status regarding this? Thanks!

**Comment by AmmarHumackicSQA** [comment]
@prih gentle bump 🙂

**Pull request closed without merging**